tutorials/aptos-tutorial.md

```diff
@@ -0,0 +1,269 @@
+in the standard library `Signer` module.  
+
+We also add `acquires InfoStore` explained before. 
+
+```move
+module Sender::UserInfo {
+    use Std::ASCII::String;
+    use Std::Signer;
+    
+    struct InfoStore has key { username: String }
+    
+    public fun get_username(account: &signer): String acquires InfoStore {
+      let account_addr = Signer::address_of(account);
+      borrow_global<InfoStore>(account_addr);
+    }
+}
+```
+
+Implementation of the setter is a bit more involved. We need to create an if-statement with two branches:
+* first for the case when there's no `InfoStore` in the global storage. We need to create one with the correct username
+* second is just to update username inside the existing `UserInfo`
+
+```move
+module Sender::UserInfo {
+    use Std::ASCII::String;
+    use Std::Signer;
+    
+    struct InfoStore has key { username: String }
+    
+    public fun get_username(account: &signer): String acquires InfoStore {
+        let account_addr = Signer::address_of(account);
+        borrow_global<InfoStore>(account_addr);
+    }
+  
+    public fun set_username(account: &signer, username: String) acquires InfoStore {
+        let account_addr = Signer::address_of(account);
+        // `exists` just to check whether resource is present in storage
+        if (!exists<InfoStore>(account_addr)) {
+            let info_store = InfoStore { username: username };
+            move_to(account, info_store);
+        } else {
+            // `borrow_global_mut` is to fetch mutable reference, we can change resources in storage that way 
+            let existing_info_store = borrow_global_mut<InfoStore>(account_addr);
+            existing_info_store.username = username;
+        }
+    }
+}
+```
+
+### Tests
+
+Now, let's write a test for our module to make sure everything works correctly. Test functions could be added anywhere, 
+but it's a nice convention to store those in the `tests/{$MODULE_NAME}Tests.move` module. 
+
+Create a `tests/` directory to the root of the package, and add `UserInfoTests.move` there. Test functions are marked 
+with `#[test]` attribute, this way Move knows it's a test and can apply special treatment:
+
+```move
+module Sender::UserInfoTests {
+    use Std::ASCII;
+  
+    use Sender::UserInfo; 
+    
+    // this named parameter to the test attribute allows to provide a signer to the test function, 
+    // it should be named the same way as parameter of the function
+    #[test(account = @0x42)]
+    fun test_getter_setter(account: signer) {
+        // ASCII::string() function allows to create a `String` object from a bytestring  
+        let username = ASCII::string(b"MyUser");
+        UserInfo::set_username(&account, username);
+      
+        // assert! macro for asserts, needs an expression and a failure error code 
+        assert!(UserInfo::get_username(&account) == username, 1);
+    }          
+}
+```
+
+Now, to run this test:
```

Comment:
Would be good to continue with:

* Convert public function to script
* Publish modules
* Run scripts

I can help with it if needed